streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -105,10 +107,10 @@ private void putAndMaybeForward(final ThreadCache.DirtyEntry entry,
                 context.setRecordContext(entry.entry().context());
                 try {
                     flushListener.apply(
-                        entry.key().get(),
-                        rawNewValue,
-                        sendOldValues ? rawOldValue : null,
-                        entry.entry().context().timestamp());
+                        new Record<>(
+                            entry.key().get(),
+                            new Change<>(rawNewValue, sendOldValues ? rawOldValue : null),
+                            entry.entry().context().timestamp()));
```

Comment:
Similar to the earlier question, does it makes sense to go ahead and add headers in now as well?